src/components/PadBtn/PadBtn.tsx

```diff
@@ -17,43 +35,143 @@ const PadBtn: React.FC<Props> = ({ input, calc, setCalc }) => {
     }
 
     if (input === "=") {
+      if (newCalc.result === 0) newCalc.result = parseFloat(mainStr);
+      setPastEquations(
+        pastEquations.concat(
```

Comment:
here you can do this way 
setPastEquations((prevState) => [...prevState, (<div> ... </div>)]
this is the correct way to update an array state
https://beta.reactjs.org/learn/updating-arrays-in-state

Reply:
It didn't work with "...prevState", but it it work with "...state". Just like in the link. 
Also, the link says it is also preferred to use concat. I do agree your suggestion fits better, but concat it does't seem do be something do avoid at all costs.